debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/structureformat/extensions/J9ObjectStructureFormatter.java

```diff
@@ -306,55 +379,142 @@ private void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPointer l
 					
 					if (isLockword) {
 						/* Print the lockword field if it is indeed the lockword for this instanceClass and we haven't printed it yet. */
-						printField = (!lockwordPrinted && (instanceClass.lockOffset().eq(superclass.lockOffset())));
+						printField = (!lockwordPrinted && (localClazz.lockOffset().eq(superclass.lockOffset())));
 						if (printField) {
 							lockwordPrinted = true;
 						}
 					}
 				}
 				
 				if (printField) {
-					printObjectField(out, tabLevel, localClazz, dataStart, superclass, result);
+					if (showNestedFields) {
+						J9ClassPointer containerClazz = null;
+						if (null == nestedClassHeirarchy) {
+							/* if nestedClassHeirarchy is null then this class is the container class */
+							containerClazz = localClazz;
+						} else {
+							containerClazz = nestedClassHeirarchy[0];
+						}
+						printNestedObjectField(out, tabLevel, localClazz, dataStart, superclass, result, address, nestingHeirarchy, containerClazz);
+					} else {
+						printObjectField(out, tabLevel, localClazz, dataStart, superclass, result, address, nestingHeirarchy);
+					}
+					
 					out.println();
 				}
 			}
 			previousSuperclass = superclass;
 		}
 	}
 	
-	public void printObjectField(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ClassPointer fromClass, J9ObjectFieldOffset objectFieldOffset) throws CorruptDataException 
+	public static void printObjectField(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ClassPointer fromClass, J9ObjectFieldOffset objectFieldOffset, long address) throws CorruptDataException {
+		printObjectField(out, tabLevel, localClazz, dataStart, fromClass, objectFieldOffset, address, null);
+	}
+	
+	public static void printObjectField(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ClassPointer fromClass, J9ObjectFieldOffset objectFieldOffset, long address, String[] nestingHeirarchy) throws CorruptDataException 
 	{
 		J9ROMFieldShapePointer fieldShape = objectFieldOffset.getField();
 		UDATA fieldOffset = objectFieldOffset.getOffsetOrAddress();
 		boolean isHiddenField = objectFieldOffset.isHidden();
-		
-		String className = J9UTF8Helper.stringValue(fromClass.romClass().className());
+		boolean containerIsFlatObject = (nestingHeirarchy != null) && (nestingHeirarchy.length > 0);
+		String className = J9ClassHelper.getName(fromClass);
 		String fieldName = J9UTF8Helper.stringValue(fieldShape.nameAndSignature().name());
 		String fieldSignature = J9UTF8Helper.stringValue(fieldShape.nameAndSignature().signature());
-		
+		boolean fieldIsFlattened = ValueTypeHelper.isFieldInClassFlattened(localClazz, fieldName);
```

Comment:
If I read the comments in `ValueTypeHelper` properly, this call is illegal if `!containerIsFlatObject`.

Reply:
I've modified `isFieldInClassFlattened` to check if VTs are enabled. It returns false if VTs are not enabled.